eventkit_cloud/ui/static/ui/app/components/StatusDownloadPage/DataPackDetails.js

```diff
@@ -34,78 +35,112 @@ export class DataPackDetails extends React.Component {
         this.setState({selectedProviders: selectedProviders});
     }
 
-    checkAll(e, checked) {
-        let stateProviders = {...this.state.selectedProviders};
-        let alteredTasks = {};
-        this.props.providerTasks.forEach((column) => {
-            if(column.display == true) {
-                let uid = column.uid;
-                alteredTasks[uid] = checked;
-            }
-        })
-        this.setState({selectedProviders: alteredTasks});
+    componentWillReceiveProps(nextProps) {
+        if(nextProps.zipFileProp != null){
```

Comment:
Could you change this to maybe a ternary in render or something else instead of storing in the state? As is, this will set state every single time the component receives props which isnt really needed.